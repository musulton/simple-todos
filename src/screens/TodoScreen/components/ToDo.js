import React from 'react'
import {View, Text, StyleSheet} from 'react-native'
import ToDoButton from "../../../shared/components/ToDoButton";

const ToDo = ({todo, toggleComplete, deleteTodo}) => (
    <View style={styles.todoContainer}>
        <Text style={[styles.todoText, todo.complete ? styles.completeTodoText : null]}>
            {todo.title}
        </Text>
        <View style={styles.buttons}>
            <ToDoButton
                name='Done'
                onPress={() => toggleComplete(todo.todoIndex)}/>
            <ToDoButton
                name='Delete'
                onPress={() => deleteTodo(todo.todoIndex)}/>
        </View>
    </View>
)
const styles = StyleSheet.create({
    todoContainer: {
        marginLeft: 20,
        marginRight: 20,
        //Mandatory style for shadowing
        backgroundColor: '#ffffff',
        borderWidth: 1,
        borderColor: '#ededed',
        paddingLeft: 14,
        paddingTop: 7,
        paddingBottom: 7,
        //Shadowing for IOS
        shadowOpacity: 0.2,
        shadowRadius: 3,
        shadowColor: '#000000',
        shadowOffset: {width: 3, height: 3},
        //Shadowing for Android
        elevation: 3,
        flexDirection: 'row',
        alignItems: 'center'
    },
    todoText: {
        fontSize: 16
    },
    completeTodoText: {
        textDecorationLine: 'line-through'
    },
    buttons: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
        alignItems: 'center'
    },
})
export default ToDo;
