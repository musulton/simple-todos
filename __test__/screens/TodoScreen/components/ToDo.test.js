import {fireEvent, render, screen} from "@testing-library/react-native";

import ToDo from "../../../../src/screens/TodoScreen/components/ToDo";

jest.mock('expo-font', () => ({
    loadAsync: jest.fn(),
    isLoaded: jest.fn().mockReturnValue(true)
}))

describe("ToDo", () => {
    test("should return correctly", () => {
        const mockProps = {
            todo: {
                title: "title",
                id: 1
            },
            onCompleted: jest.fn(),
            onDelete: jest.fn()
        }

        render(<ToDo {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should call onCompleted props when Done button is clicked", () => {
        const mockProps = {
            todo: {
                title: "title",
                id: 1
            },
            onCompleted: jest.fn(),
            onDelete: jest.fn()
        }

        render(<ToDo {...mockProps} />)

        fireEvent.press(screen.getByText(""))

        expect(mockProps.onCompleted).toHaveBeenCalledWith(mockProps.todo)
    })

    test("should call onDeleted props when Delete button is clicked", () => {
        const mockProps = {
            todo: {
                title: "title",
                id: 1
            },
            onCompleted: jest.fn(),
            onDelete: jest.fn()
        }

        render(<ToDo {...mockProps} />)

        fireEvent.press(screen.getByText(""))

        expect(mockProps.onDelete).toHaveBeenCalledWith(mockProps.todo.id)
    })
})
