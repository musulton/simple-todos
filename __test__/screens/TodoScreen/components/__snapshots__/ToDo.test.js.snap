// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo should return correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#ededed",
      "borderWidth": 1,
      "elevation": 3,
      "flexDirection": "row",
      "marginLeft": 20,
      "marginRight": 20,
      "paddingBottom": 7,
      "paddingLeft": 14,
      "paddingTop": 7,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
        },
        null,
      ]
    }
  >
    title
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "borderColor": "#ededed",
          "borderRadius": 4,
          "borderWidth": 1,
          "marginRight": 5,
          "padding": 7,
          "width": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "green",
              "fontSize": 16,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "borderColor": "#ededed",
          "borderRadius": 4,
          "borderWidth": 1,
          "marginRight": 5,
          "padding": 7,
          "width": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "rgba(175, 47, 47, 1)",
              "fontSize": 16,
            },
            undefined,
            {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
